### DISCUSSION TITLE
Broadlink riutilizzare i vecchi codici appresi sulla nuova versione

### DISCUSSION THREAD

**Discussion by nonsviluppo** [indomus/forum#1701] [HUB: Home Assistant]
Ciao a tutti, 
vi chiedo gentilmente se qualcuno potrebbe aiutarmi, ho installato  la nuova versione di broadlink , ora la mia domanda è questa :
 posso in qualche modo copiare tutti i codici dei vari device che avevo configurato  nella vecchia versione  in questa nuova senza dover far riapprendere nuovamente tutti? 
mille grazie per il vostro prezioso aiuto

**Comment by indomus** [comment]
Spiega meglio come usi i Broadlink con Home Assistant

**Comment by nonsviluppo** [comment]
Eccomi intanto grazie per la risposta, nella precedente versione andavo a compilare il file yaml dentro home assistant inserendo a mano i parametri come l'indirizzo IP dei miei tre dispositivi broadlink poi di seguito aggiungevo i comandi catturati con broadlink manager che ho installato sul PC. Uso comandi banali come accendo e spengi fancoil oppure televisore etc. Questi comandi alfanumerici una volta acquisiti li inserivo a mano. Ora nella nuova istallazione ho configurato i 3 ip che mi occorrono per individuare i miei dispositivi che sono 2 rm mini e un pro .ma seguendone varie procedure mi dice che dovrei riacquisire da capo tutti i comandi, e devo dire che ho provato ma purtroppo mi rimane difficile non so dove sbaglio. Detto questo siccome mi ero tenuto i codici che avevo acquisito tempo fa , chiedevo se fosse possibile riportarli in qualche modo .
Non so se sono stato un pochino più chiaro 
Vi ringrazio del tempo che potete dedicarmi 
Grazie di cuore

**Reply by KC-inDomus** [reply]
ora si usano telecomandi virtuali associati ai broadlink. leggi qui: https://indomus.it/guide/gestire-telecomandi-virtuali-su-home-assistant-via-broadlink-parte-1/

**Comment by nonsviluppo** [comment]
Grazie per la risposta KC , io alla fine io configuravo con la vecchia versione in questo modo il tutto poi copiando il componente smart ir nellla direcrory custom_component
 
dalla guida che ho letto  mi sembra faccia più o meno questo giusto? quindi installare il componente nativo broadlink non serve?

switch:
- platform: broadlink
  host: "192.xxxxx"
  mac: "7800000000"
  
- platform: broadlink
  host: "192."
  mac: "780000007"
  timeout: 15
  switches:
    fancoil_cucina:
      friendly_name: 'fancoil cucina'
      command_on: "JgCIAN/oFSwWgRUrFSwVLBUsFSsVgRaBFYEVghUrFisWgRUrFSwWgRUrFSwWKxUsFSsVLBWBFisVLBUsFYEVLBUsFSsVLBUsFSwVKxUsFisVLBQsFisVLBUrFS0VKxUsFSwVKxYrFSsWLBUrU="
      command_off: "JgCIAOHmFysVgRUrFiwVKxUsFSsXgBUsFSsVLBYrFYEVLBYrFYEVgRYsFYEVKxYrFisVLBUrFSwWKxUrFoEWKxUrFisWKxYrFSsVLRUrFisVLBUrFisVKxcrFSsVLBYrFSsU="

**Comment by nonsviluppo** [comment]
buongiorno a tutti.
arrivati a questo punto cosa mi consigliate smart ir oppure il componente nativo broadlink
mille grazie ancora

**Reply by KC-inDomus** [reply]
devi scegliere tu. leggi quest'altro: https://indomus.it/formazione/come-usare-gli-attuatori-broadlink-su-home-assistant/

**Comment by nonsviluppo** [comment]
Intanto grazie per le risposte l'imbarazzo della scelta anche se per tutte e tre le soluzioni ho capito che devo far riapprendere i codici e non posso copiare da nessuna parte quelli che avevo in precedenza
devo ire che questa community è fantastica complimenti
ancora grazie

**Comment by nonsviluppo** [comment]
buongiorno , le ho provate tutte ma non riesco a configurare nulla , ormai ho capito che non riesco proprio, ma guide ancora più elementari a prova di asinello ci sono?
mille grazie

**Reply by indomus** [reply]
le guide sono dettagliate. O fai capire cosa stai facendo, o altrimenti come ti si aiuta?

**Comment by nonsviluppo** [comment]
ad esempio , nella configurazione smart ir quando arrivo a questa parte 
climate:
  - platform: smartir
    name: Mitsubishi
    device_code: 1000
    controller_data: nome_della_entità_remote_broadlink
    temperature_sensor: sensor.temperature
    humidity_sensor: sensor.humidity
    power_sensor: binary_sensor.stato_accensione

già dalla parte codice io non ho il codice dei miei fancoil AERMEC pertanto non so cosa mettere  poi stessa cosa per le righe successive 
non  ho chiaro osa mettere, inoltre dove do i comandi per far accendere e spengere i fancoil?, scusate per la confusione

**Reply by KC-inDomus** [reply]
Marco (ti chiamo Marco perché non so come ti chiami), però devi sforzarti un po' di più. Se sei arrivato a questa configurazione ma non hai capito come acquisire i codici, i casi sono due:

- non hai letto con ESTREMA ATTENZIONE la guida
- non siamo capaci a spiegare le cose.

Ho un'idea sulla risposta giusta ;)

**Comment by nonsviluppo** [comment]
Ciao La risposta te la dico io ,
sicuramente ho letto con poca attenzione,, purtroppo non ho più la mente allenata allo studio visto che ho già una certa età , e si tende a far fatica,, detto questo le guide sono fatte tutte benissimo , infatti per me purtroppo è la prima volta che trovo difficoltà tutte le altre configurazioni le ho fatte seguendo le vostre guide., cmq proverò a postare passo passo i miei errori con la speranza che  la vostra pazienza nell'aiutarmi sia grande .. 
al momento vi ringrazio di tutto e sono sicuro che pian pianino ci riuscirò
grazie di tutto

**Reply by KC-inDomus** [reply]
in soldoni devi:

- acquisire tutti i codici;
- definire il tuo file json che li contenga tutti, stutturati;
- indicare tale file come riferimento nella configurazione smartir.

se segui la guida passo passo ci dovresti riuscire. avvisaci dove e quando ti blocchi e/o non capisci.

**Comment by nonsviluppo** [comment]
ancora grazie 
vi terrò aggiornati

**Comment by nonsviluppo** [comment]
Eccomi intanto grazie per la risposta, nella precedente versione andavo a compilare il file yaml dentro home assistant inserendo a mano i parametri come l'indirizzo IP dei miei tre dispositivi broadlink poi di seguito aggiungevo i comandi catturati con broadlink manager che ho installato sul PC. Uso comandi banali come accendo e spengi fancoil oppure televisore etc. Questi comandi alfanumerici una volta acquisiti li inserivo a mano. Ora nella nuova istallazione ho configurato i 3 ip che mi occorrono per individuare i miei dispositivi che sono 2 rm mini e un pro .ma seguendone varie procedure mi dice che dovrei riacquisire da capo tutti i comandi, e devo dire che ho provato ma purtroppo mi rimane difficile non so dove sbaglio. Detto questo siccome mi ero tenuto i codici che avevo acquisito tempo fa , chiedevo se fosse possibile riportarli in qualche modo .Non so se sono stato un pochino più chiaro Vi ringrazio del tempo che potete dedicarmi Grazie di cuore Inviato dal mio Galaxy
-------- Messaggio originale --------Da: inDomus ***@***.***> Data: 20/05/22  12:11  (GMT+01:00) A: indomus/forum ***@***.***> Cc: nonsviluppo ***@***.***>, Author ***@***.***> Oggetto: Re: [indomus/forum] Broadlink riutilizzare i vecchi codici appresi sulla nuova versione (Discussion #1701) 
Spiega meglio come usi i Broadlink con Home Assistant
